analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -30,10 +30,15 @@ internal sealed class Binary : BranchingProcessor<IBinaryOperationWrapper>
     protected override SymbolicConstraint BoolConstraintFromOperation(SymbolicContext context, IBinaryOperationWrapper operation) =>
         BinaryConstraint(operation.OperatorKind, context.State[operation.LeftOperand], context.State[operation.RightOperand]);
 
-    protected override ProgramState LearnBranchingConstraint(ProgramState state, IBinaryOperationWrapper operation, bool falseBranch) =>
-        operation.OperatorKind.IsAnyEquality()
-            ? LearnBranchingConstraint<ObjectConstraint>(state, operation, falseBranch) ?? LearnBranchingConstraint<BoolConstraint>(state, operation, falseBranch) ?? state
-            : state;
+    protected override ProgramState LearnBranchingConstraint(ProgramState state, IBinaryOperationWrapper operation, bool falseBranch)
```

Comment:
I got the point. Maybe it is enough for you to have a look at this commit https://github.com/SonarSource/sonar-dotnet/pull/6864/commits/ee4465effb40252d93a76041443743d25f18ce25. It shows, the fix and there is only one test affected. If you think this is enough for you to see, I would like to keep it as part of the PR, otherwise, I would revert that commit and put it in another PR.

Reply:
I'd like the master commit history to be easier to understand too.

Please revert it and cherry-pick for another PR.